src/bin/ion/commands/beta/generate/templates/rust/struct.templ

```diff
@@ -29,106 +31,52 @@ pub mod  {{ target_kind_name | snake }} {
-                        return validation_error(format!(
-                           "Expected list, found {} while reading {{ target_kind_name }}.", reader.ion_type().unwrap()
-                        ));
-                    }
-                {% elif abstract_data_type["Sequence"].sequence_type == "SExp" %}
-                    if reader.ion_type() != Some(IonType::SExp) {
-                       return validation_error(format!(
-                           "Expected sexpression, found {} while reading {{ target_kind_name }}.", reader.ion_type().unwrap()
-                       ));
-                    }
-                {% endif %}
-                 reader.step_in()?;
-
-                 abstract_data_type.value = {
-                     let mut values = vec![];
-
-                     while reader.next()? != StreamItem::Nothing {
-                        {% if abstract_data_type["Sequence"].element_type | is_built_in_type == false %}
-                            values.push({{ abstract_data_type["Sequence"].element_type }}::read_from(reader)?);
-                        {% else %}
-                            values.push(reader.read_{% if fields[0].isl_type_name == "symbol" %}symbol()?.text().unwrap(){% else %}{{ abstract_data_type["Sequence"].element_type | lower | replace(from="string", to ="str") }}()?{% endif %}{% if abstract_data_type["Sequence"].element_type | lower== "string" %} .to_string() {% endif %});
+                        {% endfor %}
+                     _ => {
+                        {% if abstract_data_type["Structure"] %}
+                            return validation_error(
+                                "Can not read field name:{{ field.name }} for {{ target_kind_name }} as it doesn't exist in the given schema type definition."
+                            );
                         {% endif %}
+                     }
                     }
-                    values
-                 };
-                 reader.step_out()?;
-            {% else %}
-                return validation_error("Can not resolve read API template for {{ target_kind_name }}");
-            {% endif %}
-         Ok(abstract_data_type)
+                }
+            }
+            reader.step_out()?;
+            Ok(abstract_data_type)
         }
 
         pub fn write_to<W: IonWriter>(&self, writer: &mut W) -> SerdeResult<()> {
-            {% if abstract_data_type == "Value" %}
-                {% for field in fields %}
-                    {% if field.value_type | is_built_in_type == false  %}
-                        self.{{ field.name | snake }}.write_to(writer)?;
+            writer.step_in(IonType::Struct)?;
+            {% for field in fields %}
+            writer.set_field_name("{{ field.name }}");
+                {% if field.value_type | is_built_in_type == false %}
+                    {% if field.value_type is containing("Vec") %}
+                        {{ util_macros::write_as_sequence(field=field) }}
```

Comment:
🗺️ PR tour) This statement is modified to write fields that are of type `Vec` using util_macros `write_as_sequence`. Previously it was not possible for a field in struct to be of sequence type(e.g. It would be inside some nested type `NestedtypeX`). But with this change it is possible to have fields with sequence type (e.g. `Vec<i64>`).